docs!: Docs information architecture refactor

bpmutter opened the pull request:
## Summary

This document contains proposed changes to the information architecture in the ESLint documentation website, <https://eslint.org/docs>.

On a high level, the proposed information architecture breaks the content into the following sections:

- Use ESLint in Your Project: A refactor of the current [User Guide](https://eslint.org/docs/latest/user-guide/)
- Extend ESLint: A refactor of the current [Developer Guide](https://eslint.org/docs/latest/developer-guide/), with some content moved to the section Maintain ESLint and Community Contributions
- Maintain ESLint: Expansion of current [Maintainer Guide](https://eslint.org/docs/latest/maintainer-guide/), with some content taken from the current ‘Developer Guide’.
- Community Contributions: Expansion of the current [Contributing section](https://eslint.org/docs/latest/developer-guide/contributing/) of the Developer Guide.

## Related Issues

https://github.com/eslint/eslint/issues/16365

nzakas reviewed (commented):
Thanks for putting this together. I left a few comments.

It would also be helpful to include a simplified new documentation index that is just bullets and sub-bullets that shows the intended complete view of the left side of the docs. It’s a little difficult to envision it from the way this is structured currently.

bpmutter commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,252 @@
+  - Take most of content from the section [Processors in Plugins section on Work with Plugins Page](https://eslint.org/docs/latest/developer-guide/working-with-plugins#processors-in-plugins)
+- [RENAME PAGE] Share Configurations
+  - Current page: [Shareable Configs](https://eslint.org/docs/latest/developer-guide/shareable-configs)
+- [NO CHANGE TO PAGE] Node.js API
```
i think keeping it here makes sense b/c even if it's not being used by a plugin directly, other it's useful in other "extender" scenarios, such as:

- developing an integration
- testing an extension (plugin, custom formatter, etc). when i wrote that blog post about custom formatters, i actually used the Node.js API  to test the formatter ([link](https://github.com/eslint/eslint.org/pull/333)).

nzakas replied:
Ah okay, that makes sense. I was thinking of extender as purely plugin developers but grouping in those others makes this the right spot.

bpmutter commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,252 @@
+  - [MOVE PAGE] Governance
+    - Current page: [Governance](https://eslint.org/docs/latest/maintainer-guide/governance)
+
+### [NEW SECTION] Community Contributions
```
makes sense to me

bpmutter commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,252 @@
+
+## Open Questions
+
+1. Is there a way to automatically change all the internal links that'll be broken by these updates?
```
ok, that'll take some manual work, but shouldn't be an issue. plus, i believe 11ty will show an error if one of these internal links breaks, so there's low risk of a broken link entering production.

bpmutter replied:
> It would be helpful to include a table that shows current URLs and planned replacement URLs for each.

ok, will add

mdjermanovic replied:
> plus, i believe 11ty will show an error if one of these internal links breaks, so there's low risk of a broken link entering production.

I don't think 11ty does that, and I don't think we currently have any checks for broken internal links.

bpmutter replied:
ah, that's unfortunate. even if the updates are manual, i don't think it should be too bad with some find/replacing. 

however, i would like to at least find a semi-automated way to check that no broken internal links enter production. will dig into this some more.

bpmutter commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,252 @@
+## Open Questions
+
+1. Is there a way to automatically change all the internal links that'll be broken by these updates?
+1. Can you create a side navigation folder that is not also a page?
```
i mean a sidebar container that displays a name and has child elements, but isn't a page with content. 

in Docusaurus, this is called a [container](https://docusaurus.io/docs/sidebar/items#sidebar-item-category)

nzakas replied:
Ah gotcha. The answer is yes, those don’t need to have landing pages. We have them currently because there are existing links out there that point to those pages but we don’t actually link them from the sidebar.

nzakas replied:
Oops sorry, misunderstood this.

So we have the top level items like User Guide that aren’t linked to anything. We don’t have the ability to do the same anywhere else in the sidebar, so we always need a landing page for anything other than that top level. This was a design decision to make it clear what was clickable and what wasn’t.

bpmutter replied:
ok thanks for clarifying this. will keep this consideration in mind for the IA changes. right now going thru what's already written so see if any updates necessary. will add if so.

bpmutter commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,252 @@
+
+1. Is there a way to automatically change all the internal links that'll be broken by these updates?
+1. Can you create a side navigation folder that is not also a page?
+1. How to make section landing pages navigable from the side table of contents? (For example, the proposed page "Use ESLint in Your Project landing page")
```
sure. i think it's best illustrated by an example. in the current ESLint site, the page https://eslint.org/docs/latest/user-guide/ cannot be accessed from the left-hand side navigation.

nzakas replied:
My previous comment actually makes more sense here. 😄 

> We have them currently because there are existing links out there that point to those pages but we don’t actually link them from the sidebar.

This was also a design decision because, for accessibility purposes, there isn’t a clear way to indicate that something is both a link and expandable. If we didn’t already have landing pages for user guide, etc., then we wouldn’t make them.

bpmutter replied:
i'll add a note about this to the RFC as well

bpmutter commented:
> t would also be helpful to include a simplified new documentation index that is just bullets and sub-bullets that shows the intended complete view of the left side of the docs. It’s a little difficult to envision it from the way this is structured currently.

sure will add

bpmutter commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,252 @@
+
+- Use ESLint in Your Project: A refactor of the current [User Guide](https://eslint.org/docs/latest/user-guide/)
+- Extend ESLint: A refactor of the current [Developer Guide](https://eslint.org/docs/latest/developer-guide/), with some content moved to the section Maintain ESLint and Community Contributions
+- Maintain ESLint: Expansion of current [Maintainer Guide](https://eslint.org/docs/latest/maintainer-guide/), with some content taken from the current ‘Developer Guide’.
```
> Also not sure how useful the maintainer docs actually are.

let's address this in Phase 5: “Maintain ESLint” documentation update

> Governance could be moved to the Contributor section?

makes sense to me. moving.

nzakas replied:
After thinking about this some more, I think we are actually missing a section: "Integrating ESLint". This is distinct from "Extending ESLint," which is about creating plugins, processors, shareable configs, etc. "Integrating ESLint" is about using the ESLint API to include ESLint somewhere else: in VS Code, `gulp-eslint`, etc.

What do you think about adding an "Integrating ESLint" section that includes both the Integrations page and the Node.js API page as a starting point?

bpmutter replied:
i think i addressed this comment here - https://github.com/eslint/rfcs/pull/97#discussion_r1014672395

bpmutter commented:
@nzakas responded to and implemented your feedback!

nzakas commented:
@bpmutter Hmmm, I’m not seeing any changes, did you push your latest commit?

bpmutter commented:
>  @bpmutter Hmmm, I’m not seeing any changes, did you push your latest commit?
@nzakas sorry, didn't push yesterday. just did now. you can find the content there.

mdjermanovic commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,351 @@
+  - Propose a New Rule
+  - Propose a Rule Change
+  - Request a Change
+  - Work on Issues
+  - Report Security Vulnerabilities
+  - Submit a Pull Request
```
"Submit a Pull Request" should be right after "Work on Issues", as they're closely related (someone working on an issue will eventually submit a pull request).

bpmutter replied:
i'll shift that page up 1 in the table of contents

mdjermanovic commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,351 @@
+  - Submit a Pull Request
+  - Governance
+- Maintain ESLint (landing page)
+  - Architecture
+  - Set up a Development Environment
+  - Run the Tests
```
I think all these three should be in Contribute to ESLint.

bpmutter replied:
yeah, that makes sense. will move. 

i'd initially put these pages in the maintainer section before we'd conceived of the "contributor" persona

mdjermanovic commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,351 @@
+  - Custom Processors
+  - Share Configurations
+  - Node.js API Reference
+- Contribute to ESLint (landing page)
```
Does "landing page" mean that "Contribute to ESLint" will be a link in the left-hand side navigation? (pointing to a page with [this content](https://eslint.org/docs/latest/developer-guide/contributing/)).

bpmutter replied:
it will be that page with the same content, but given the current way the sidebar works, there will not be a clickable link there. this was a decision made in the past for accessibility purposes. 

nicholas and i were discussing the matter in [this comment thread](https://github.com/eslint/rfcs/pull/97#discussion_r1007459580).

mdjermanovic replied:
Okay, then we should check if there's any important information that only exists on this page because the page (and therefore its content) will effectively be hidden since it won't be in the navigation anymore.

In particular:

 * We should probably have a new page for the [Code of Conduct](https://eslint.org/docs/latest/developer-guide/contributing/#read-the-code-of-conduct) section. Perhaps we could also keep the first paragraph from the current [contributing](https://eslint.org/docs/latest/developer-guide/contributing/) page on this page?
* The CLA section could be merged into the Submit a Pull Request page. There's actually already something [here](https://eslint.org/docs/latest/developer-guide/contributing/pull-requests#step-7-send-the-pull-request), but with fewer details.

This RFC already proposes a new page for [reporting a security vulnerability](https://eslint.org/docs/latest/developer-guide/contributing/#reporting-a-security-vulnerability) so that's covered. Other sections do not seem to provide important additional info.

bpmutter replied:
added this content in these 2 PRs:
- 2ca6ee9
- d91afa0

bpmutter commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,351 @@
+  - Current page: [Working with Plugins](https://eslint.org/docs/latest/developer-guide/working-with-plugins)
+  - As noted below, convert the “Processors in Plugins” section into separate page “Custom Processors
+  - Add overview-style content at the top of the page explaining that this page explains how to bundle together the various parts of a plugin, and then publish it.
+- [RENAME PAGE] Custom Rules
+  - Current page: [Working with Rules](https://eslint.org/docs/latest/developer-guide/working-with-rules)
```
thanks for calling this out. i think keeping the page as is for now is OK for the initial information architecture changes.

i would imagine that anyone wanting to contribute to the core rules could figure out what to do. and anyone looking to create a custom rule could ignore the content about adding a core rule.

with that being said, i think it'd make sense as a fast follow item to split up the content between these 2 pages. 

i would like to keep these tasks separate b/c i want to keep the information architecture updates very narrowly focused on *moving and renaming* pages and sections, not creating new content. this is to streamline the review process and commit history.

mdjermanovic replied:
> i would imagine that anyone wanting to contribute to the core rules could figure out what to do. and anyone looking to create a custom rule could ignore the content about adding a core rule.

I agree that both an Extender and a Community Contributor will most likely figure out what's relevant to them when they get to the document (though it would be nice to make this clear at some point, as you suggested).

I am concerned about how a new Community Contributor who wants to create a new core rule or change an existing one will get to the document, because "Extend ESLint > Custom Rules" item doesn't sound applicable to what they want to do.

nzakas replied:
> I am concerned about how a new Community Contributor who wants to create a new core rule or change an existing one will get to the document, because "Extend ESLint > Custom Rules" item doesn't sound applicable to what they want to do.

I'm not too concerned about this. We can very easily include a link from the contributing section that says "Core rules are written in the same way as custom rules with the following exceptions."

mdjermanovic replied:
If we plan to move and rename pages before creating new content, after we move and rename pages as proposed it won't be clear for contributors where to look for information on how to create or change a core rule.

https://github.com/eslint/rfcs/blob/c2641344b8c5f3d44f735182e3801aa380dabe8f/designs/2022-docs-information-architecture-update/README.md?plain=1#L181-L200

bpmutter replied:
i am going to do a bit of looking at splitting this page up as part of the IA refactor per your points @mdjermanovic. 

as mentioned above, it was my intention to not do any meaningful content refactors as part of the  the IA update. however, this might be an exception to that principle.

plus we could do the page split before the IA-refactor, and then move those 2 separate pages as part of the IA refactor.

bpmutter replied:
see this comment for proposed solution here - https://github.com/eslint/rfcs/pull/97#issuecomment-1304605294

mdjermanovic replied:
> see this comment for proposed solution here - [#97 (comment)](https://github.com/eslint/rfcs/pull/97#issuecomment-1304605294)

The comment seems to have disappeared for some reason. I can see it in my email, but can't find it here and this link doesn't work.

bpmutter replied:
yeah i don't see it either. this is strange and frustrating. i spent a fair bit of time coming up with that suggestion, so don't want to rethink and rewrite if possible. can you paste it from your email here, and respond to it? (i didn't find it in mine)

btmills replied:
This one? Pulled from my email:

> @nzakas and @mdjermanovic,
>
> i think i got the solution for this problem of how to split the rules documentation between core rules for contributors and custom rules for extenders.
>
> what i propose is:
>
> 1.  keep the majority of the rule creation content on the "Custom Rules" page in the Extender documentation section. remove content about creating core rules. note somewhere on this page that if you want to create a core rule, refer to the Core Rules documentation (expanded on in next point).
> 2. create a new "Core Rules" page in the Contributor documentation section. extract all content specifically about creating core rules into this page. This page should, as Nicholas proposed, point to the "Custom Rules" page for most of the rule creation documentation.
> 3. do this work as a fast follow to the initial IA migration. while there could certainly be some confusion about how to work on custom rules between the initial migration and the completion of this work, i don't think it'd be a significant issue for the following reasons:
>
> - i don't think many people are working on core rules at a given moment
> - those people can probably figure out what they need to from the custom rules documentation.
> - if those people run into any issues, they can ask in the Discord, and we can point them in the right direction.
> - the amount of time when this is a potential issue would be pretty short, especially if we prioritize the reviews on the content splitting the pages, so that the changes get moved promptly.

nzakas replied:
Here's the thing: the number of times someone will write a new core rule from scratch is tiny compared to the number of times that someone will create a custom rule. Given that, I think we should optimize for the custom rule use case and extend out from that for core rules. I also look forward to the day when there are no core rules, so also don't want to overly-optimize for that.

That said, I'm 👍 to @bpmutter's suggestion as posted above by @btmills

bpmutter replied:
@btmills thank you for finding and posting that!

@mdjermanovic how you feel abt the above suggestion?

mdjermanovic replied:
https://github.com/eslint/rfcs/pull/97#discussion_r1017440912 sounds good to me :+1:

bpmutter replied:
added to the Documentation update project as part of the Information Architecture update phase 2 as a fast follow to initial updates.

bpmutter commented:
@mdjermanovic responded to your feedback!

nzakas commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,352 @@
+
+- **The User**: Someone who wants to use ESLint as it currently exists, including plugins. The proposed 'Use ESLint in Your Project' section addresses this persona's needs.
+- **The Extender**: Someone who wants to extend the functionality of ESLint by creating a plugin, custom formatter, custom parser, sharable configuration, etc. The proposed 'Extend ESLint' section addresses this persona's needs.
+- **The Maintainer**: Someone who wants to contribute to the core ESLint project. The proposed 'Maintain ESLint' section addresses this persona's needs.
```
Per my previous comment:

**The Integrator**: Someone who wants to embed ESLint in another application.

bpmutter replied:
so for me, the Node.js API makes sense for 'the integrator' persona. though i imagine "extenders" would also use it when testing their extensions.  

however, i see the [integrations page](https://eslint.org/docs/latest/user-guide/integrations) as being more oriented for end users (the user persona). 

per these points, i think it makes sense to keep integrations as they are on this outline. 

though of course, open to have my mind changed on this 😄

bpmutter replied:
thinking a little more on this, rather than breaking of "the integrator" as a separate persona, perhaps we can consider the integrator as a "sub-persona" of the "extender", in the same way that a plugin developer and a custom rule developer are "sub-personas" of the extender.

per this, it could make sense to add a 'Create Integrations' page within the 'Extend ESLint' section. of course, adding this page wouldn't be part of the IA project, but could be a good inclusion in Phase 4: Extend ESLint documentation update.

@nzakas what do you think about this option?

nzakas replied:
I just don't agree that integrator is a sub-persona of extender. To me, these are two very different use cases with the only commonality being that they both write JavaScript code to work with ESLint.

* Extender - Uses the core of ESLint and creates new rules, parsers, etc. that ESLint consumes during its normal course of execution. The goal is to extend ESLint with more functionality.
* Integrator - Uses ESLint to extend some other program, such as an IDE, CI system, or build system. The goal here is not to add more functionality to ESLint, but rather to include ESLint in some other program.

I'm not seeing any overlap between these two use cases. Is there some overlap that I'm missing?

nzakas replied:
Oh, missed the last point: totally fine with keeping "Integrations" where they are in the outline. I just feel like we need another section to help people embed ESLint elsewhere.

bpmutter replied:
@nzakas thank you for clarifying there. i see what you're saying. makes sense to me as well.

let's add an "Integrate ESLint" section to the docs.

with that being said, i'd rather not include it in this first pass of information architecture updates b/c as it stands right now, the section would just contain the "Node.js API". i'd like to avoid net-new content as part of the information architecture updates, to keep this phase of the project limited in scope. 

per this, i've made a new phase (6) of the [documentation update project](https://github.com/eslint/eslint/issues/16365) to cover the integrator persona. and for the time being, i'd like to keep the Node.js API reference in the "Extend ESLint" section.

if you're good with this approach, i'll also update the RFC document to state that future documentation updates will add "integrator" persona docs that'll include the Node.js API reference.

nzakas replied:
Yup, that makes total sense to me. Thanks!

bpmutter replied:
added line about it in this commit - https://github.com/eslint/rfcs/pull/97/commits/dffd504d7608517334d8b60c6c2a4e67d7303374 

and have added phase to document integrator persona in the [documentation update project](https://github.com/eslint/eslint/issues/16365).

resolving this conversation since it seems we're aligned on the next steps here @nzakas.

nzakas commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,352 @@
+
+### HTTP Redirects
+
+The proposed changes would include the following HTTP redirects from current content locations to the locations proposed in the above outline.
```
Thanks for putting this together. Some thoughts on the URL structure:

* There's an inconsistency between `/users`, `/extend`, and `/maintain`. It seems like we should be using the same word forms throughout. So, `/users`, `/extenders`, `/maintainers` or `/use`, `/extend`, `/maintain`, although I think I'd prefer `/using`, `/extending` and `/maintaining`. What do you think?
  * Whatever we decide, we should make it consistent all the way down so `/using/configuring` or `/use/configure` rather than `/users/configure`.
* I'm not sure if it's intentional or not, but there is a mix of `/users` and `/user` in the table below.

btmills replied:
Given the preference for removing avoiding gerunds, I think I'd lean toward for `/:persona/:topic`, e.g. `/contributors/pull-requests`, but I'd also be fine with the verb forms as suggested above.

bpmutter replied:
i absolutely agree with the point about consistency. good call out. 

i don't have a strong preference for the URL slug pattern.

`/:persona/:topic` as @btmills proposes has a 'RESTful' eloquence to it. however, the `:persona` could get a bit awkward with the persona names being a little clunky, such as `extenders` or `integrators`. but i think this is OK given that the context is a URL slug, not the text copy. just wanted to call this out. 

and to do verbs instead of persona (i.e `use`, `extend`, etc.) seems fine as well. though to have `use` in the slug in particular seems unclear b/c in a sense, everyone is using ESLint. but agian, it's in a URL slug, not the copy of the text, so i don't think the nuance here is too important.

so i guess tl;dr both options are fine w/ minor tradeoffs. 

i think i have a slight preference for the `:persona` based structure, but would defer to the core maintainers if you feel otherwise.

nzakas replied:
Yeah, I feel like `/:persona` may be a bit too meta -- this seems like a leaky abstraction to me.

I think something around actions makes the most sense, though I do wonder if maybe we decouple those actions from the personas. For instance, what if a user came in and they went through the following pages:

1. `/use/setup` or `/use/install`
2. `/use/configure`

Although `/use/core-concepts` seems a bit weird to me.

@eslint/eslint-tsc what do you think?

btmills commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,352 @@
+  - There should probably be a future unit of work to refactor this page.
+  - However, it’s not included in the information architecture updates.
+- [MOVE & REFACTOR PAGE] Set up Development Environment
+  - Consolidate the pages [Getting the Source Code](https://eslint.org/docs/latest/developer-guide/source-code) and [Set up a Development Environment](https://eslint.org/docs/latest/developer-guide/development-environment)
```
Good idea to combine these.

https://eslint.org/docs/latest/developer-guide/contributing/pull-requests links to dev environment setup, so should this combined page be part of the Contribute to ESLint section instead, either within or immediately after Submit a Pull Request?

*Edit: The sidebar summary below shows this page in the contributor guide. If that's been updated more recently, I agree :+1: I also noticed the location of the release guide is inconsistent.*

bpmutter replied:
looks like the sidebar summary was correct and things were added incorrectly over here. sorry for confusion. fixing now.

btmills commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,352 @@
+
+### Why Rename Pages with Active Verb Phrases?
+
+This outline proposes renaming various titles to use active verb phrases instead of the current gerund-based page name structure (ex. changing a page title from “Managing Issues” → “Manage Issues”). This is following a general technical writing best practice of avoiding gerunds in titles. For more information, refer to [this blog post about the subject](https://www.google.com/url?q=https://www.wr ...
```
Thanks for sharing this link! I was going to leave a comment at the top asking to learn more, but you anticipated that curiosity 🧠

bpmutter replied:
i had a feeling someone would comment 😃

bpmutter commented:
>> I am concerned about how a new Community Contributor who wants to create a new core rule or change an existing one will get to the document, because "Extend ESLint > Custom Rules" item doesn't sound applicable to what they want to do.
>
> I'm not too concerned about this. We can very easily include a link from the contributing section that says "Core rules are written in the same way as custom rules with the following exceptions."

@nzakas and @mdjermanovic, 

i think i got the solution for this problem of how to split the rules documentation between core rules for contributors and custom rules for extenders.

what i propose is:

1. keep the majority of the rule creation content on the "Custom Rules" page in the Extender documentation section. remove content about creating core rules. note somewhere on this page that if you want to create a core rule, refer to the Core Rules documentation (expanded on in next point).
2. create a new "Core Rules" page in the Contributor documentation  section. extract all content specifically about creating core rules into this page. This page should, as Nicholas proposed, point to the "Custom Rules" page for most of the rule creation documentation.
3. do this work as a fast follow to the initial IA migration. while there *could* certainly be some confusion about how to work on custom rules between the initial migration and the completion of this work, i don't think it'd be a significant issue for the following reasons:
  - i don't think many people are working on core rules at a given moment
  - those people can probably figure out what they need to from the custom rules documentation. 
  - if those people run into any issues, they can ask in the Discord, and we can point them in the right direction.
  - the amount of time when this is a potential issue would be pretty short, especially if we prioritize the reviews
     on the content splitting the pages, so that the changes get moved promptly.

bpmutter commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,352 @@
+  - Current page: [Managing Issues](https://eslint.org/docs/latest/maintainer-guide/issues)
+- [MOVE & RENAME PAGE] Review Pull Requests
+  - Current page: [Reviewing Pull Requests](https://eslint.org/docs/latest/maintainer-guide/pullrequests)
+- [MOVE & RENAME PAGE] Manage Releases
+  - Current page: [Managing Releases](https://eslint.org/docs/latest/maintainer-guide/releases)
```
yep, same mistake as the one [noted in this comment](https://github.com/eslint/rfcs/pull/97#discussion_r1014684139), sorry for confusion.

bpmutter commented:
@nzakas, @btmills and @mdjermanovic,

i believe i have addressed all of the currently open comment threads. there's *a lot* going on in this PR, so i'm summarizing the open questions here and where they stand.

could you please resolve the conversation threads that you believe to be closed so we can keep moving the RFC towards acceptance?

- [ ] @btmills i fixed a mistake misplacing where certain pages are supposed to be moved to address the following 2 comments. i believe both of these threads can be closed.
  - https://github.com/eslint/rfcs/pull/97#discussion_r1012759811
  - https://github.com/eslint/rfcs/pull/97#discussion_r1014684765
- [ ] @nzakas i responded to your thoughts about the proposed 'integrator persona' in this comment - https://github.com/eslint/rfcs/pull/97#discussion_r1012138932. could you take a look and respond there?
- [ ] @nzakas and @btmills i responded to the thread about the URL paths - https://github.com/eslint/rfcs/pull/97#discussion_r1012146822. could you take a look and respond there?
- [ ] @nzakas and @mdjermanovic i responded to the conversation about how to handle the rules page in this comment - https://github.com/eslint/rfcs/pull/97#discussion_r1008932638. please let me know what you think of the proposal, and if you're ok with this plan, i'll add it to the outline.

ok, i think i covered all the major in this comment, but if there's anything else, of course let me know and we'll address it 😄

mdjermanovic commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,358 @@
+- [MOVE, EXPAND & RENAME PAGE] Submit a Pull Request
+  - Current page: [Pull Requests](https://eslint.org/docs/latest/developer-guide/contributing/pull-requests)
+  - Add the content from [Signing the CLA section](https://eslint.org/docs/latest/developer-guide/contributing/#signing-the-cla) from the Contributing overview page.
+  Expand the content in [Step 7: send the pull request](https://eslint.org/docs/latest/developer-guide/contributing/#signing-the-cla) with this content.
```
```suggestion
  Expand the content in [Step 7: send the pull request](https://eslint.org/docs/latest/developer-guide/contributing/pull-requests#step-7-send-the-pull-request) with this content.
```

mdjermanovic commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,358 @@
+  - Run the Tests
+  - Work on Issues
+  - Submit a Pull Request
+  - Code of Conduct
```
Perhaps "Code of Conduct" should be the first item under "Contribute to ESLint", as it's now on the [contributing](https://eslint.org/docs/latest/developer-guide/contributing/) page?

mdjermanovic commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,358 @@
+    - Configure Plugins
+    - Ignore Code
+  - Command Line Interface Reference
+  - Rules Reference
+  - Integrations
```
```suggestion
  - Rules Reference
  - Formatters Reference
  - Integrations
```

I think "Formatters Reference" item is missing here.

nzakas commented:
**TSC Summary:** The last point to address here is how to move forward with URL structure. Possible options are 1) persona/action such as users/configure, 2) use case/action such as use/configure, 3) using/configuring

**TSC Question:** Which format do we want to use?

mdjermanovic commented:
In yesterday's TSC meeting we agreed to use `/:verb/:subject` for the URL format. In the options above, that's "2) use case/action such as use/configure".

If any path looks weird with this format, we can consider changing its URL by changing the `:subject` part (e.g., `/integrate/nodejs-api` could be `/integrate/using-api`), or by inserting `/learn/` in the middle (e.g., `contribute/architecture` could be `contribute/learn/architecture`), or in some other way so that it reads better.

bpmutter commented:
> In yesterday's TSC meeting we agreed to use /:verb/:subject for the URL format. In the options above, that's "2) use case/action such as use/configure".

thank you for the update! i'll update the URL paths to follow this pattern. will let the TSC know when that's done.

bpmutter commented:
ok, @nzakas i believe i have updated the URL paths in all parts where it was necessary. also did final review for copy changes, and everything seems to be ready for (final?) re-review.

nzakas reviewed (approved):
I think we are in pretty good shape now. Moving to final commenting.

mdjermanovic commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,358 @@
+
+| Current URL | Proposed URL |
+| :---: | :---: |
+| <https://eslint.org/docs/latest/user-guide/> | <https://eslint.org/docs/latest/use/core-concepts> |
```
```suggestion
| <https://eslint.org/docs/latest/user-guide/> | <https://eslint.org/docs/latest/use/> |
```

I think this was a typo.

mdjermanovic commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,358 @@
+| <https://eslint.org/docs/latest/extend/working-with-rules> | <https://eslint.org/docs/latest/extend/custom-rules> |
+| <https://eslint.org/docs/latest/developer-guide/working-with-plugins> | <https://eslint.org/docs/latest/extend/plugins> |
+| <https://eslint.org/docs/latest/developer-guide/working-with-custom-formatters> | <https://eslint.org/docs/latest/extend/custom-formatters> |
+| <https://eslint.org/docs/latest/developer-guide/working-with-custom-parsers> | <https://eslint.org/docs/latest/custom-parsers> |
```
```suggestion
| <https://eslint.org/docs/latest/developer-guide/working-with-custom-parsers> | <https://eslint.org/docs/latest/extend/custom-parsers> |
```

mdjermanovic commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,358 @@
+| <https://eslint.org/docs/latest/developer-guide/working-with-plugins> | <https://eslint.org/docs/latest/extend/plugins> |
+| <https://eslint.org/docs/latest/developer-guide/working-with-custom-formatters> | <https://eslint.org/docs/latest/extend/custom-formatters> |
+| <https://eslint.org/docs/latest/developer-guide/working-with-custom-parsers> | <https://eslint.org/docs/latest/custom-parsers> |
+| <https://eslint.org/docs/latest/developer-guide/nodejs-api> | <https://eslint.org/docs/latest/extend/nodejs-api> |
```
```suggestion
| <https://eslint.org/docs/latest/developer-guide/nodejs-api> | <https://eslint.org/docs/latest/integrate/nodejs-api> |
```

Since this will later be moved to the Integrate ESLint section, perhaps we could use the `/integrate/` path from the start?

bpmutter replied:
sure, i don't see any harm in doing this.

mdjermanovic commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,358 @@
+| <https://eslint.org/docs/latest/developer-guide/nodejs-api> | <https://eslint.org/docs/latest/extend/nodejs-api> |
+| <https://eslint.org/docs/latest/developer-guide/contributing/> | <https://eslint.org/docs/latest/contribute/> |
+| <https://eslint.org/docs/latest/developer-guide/contributing/reporting-bugs> | <https://eslint.org/docs/latest/contribute/report-bugs> |
+| <https://eslint.org/docs/latest/developer-guide/contributing/new-rules> | <https://eslint.org/docs/latest/contribute/new-rule> |
+| <https://eslint.org/docs/latest/developer-guide/contributing/rule-changes> | <https://eslint.org/docs/latest/contribute/rule-change> |
+| <https://eslint.org/docs/latest/developer-guide/contributing/changes> | <https://eslint.org/docs/latest/contribute/change> |
```
```suggestion
| <https://eslint.org/docs/latest/developer-guide/contributing/new-rules> | <https://eslint.org/docs/latest/contribute/propose-new-rule> |
| <https://eslint.org/docs/latest/developer-guide/contributing/rule-changes> | <https://eslint.org/docs/latest/contribute/propose-rule-change> |
| <https://eslint.org/docs/latest/developer-guide/contributing/changes> | <https://eslint.org/docs/latest/contribute/request-change> |
```

I think these should have propose/request. `contribute/new-rule` sounds more like implementing a new rule.

mdjermanovic commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,358 @@
+| <https://eslint.org/docs/latest/developer-guide/source-code> |  <https://eslint.org/docs/latest/contribute/development-environment> |
+| <https://eslint.org/docs/latest/developer-guide/development-environment> |  <https://eslint.org/docs/latest/contribute/development-environment> |
+| <https://eslint.org/docs/latest/developer-guide/unit-tests> | <https://eslint.org/docs/latest/contribute/tests> |
+| <https://eslint.org/docs/latest/extend/working-with-rules> | <https://eslint.org/docs/latest/extend/custom-rules> |
```
```suggestion
| <https://eslint.org/docs/latest/developer-guide/working-with-rules> | <https://eslint.org/docs/latest/extend/custom-rules> |
```

mdjermanovic commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,358 @@
+
+### [REFACTOR SECTION]  Extend ESLint
+
+- [NO CHANGE] Extend ESLint landing page
+  - Current page:  [Developer Guide](https://eslint.org/docs/latest/developer-guide/)
+  - Hidden in sidebar
```
We should move some content from this page to the contributing landing page, as it's currently a mix of both.

bpmutter replied:
good point. i can add a note about this.

bpmutter commented on designs/2022-docs-information-architecture-update/README.md:
```diff
@@ -0,0 +1,358 @@
+- [NEW PAGE(S)] Any other technical pages for maintainers
+  - I don’t have any thoughts on what these should be right now. I would defer to the core maintainers on what they think other maintainers should know.
+  - I think that these additional technical pages would make sense to include in **Phase 5: "Maintain ESLint" documentation updates**of [the documentation update project](https://github.com/eslint/eslint/issues/16365).
+- [MOVE PAGE] Run the Tests
+  - Current page: [Unit Tests](https://eslint.org/docs/latest/developer-guide/unit-tests)
```
that makes sense to me

bpmutter commented:
@mdjermanovic ready for re-review

mdjermanovic reviewed (approved):
Looks good to me! Great idea to reorient around the core personas and split the Developer Guide into Contribute to ESLint and Extend ESLint.

nzakas commented:
All right, it looks like we are ready to move forward with this. Thanks everyone!
